src/link/output.cpp

```diff
@@ -558,6 +558,19 @@ static void writeSym() {
 		for (uint32_t bank = 0; bank < sections[type].size(); bank++)
 			writeSymBank(sections[type][bank], type, bank);
 	}
+
+	// Output the exported numeric constants
+	static std::vector<Symbol *> constants; // `static` so `sym_ForEach` callback can see it
```

Comment:
```suggestion
	std::vector<Symbol *> constants;
```
The callback can see it fine, and the array doesn't need to linger past the callback's lifetime, so we're good to make this automatic.

Reply:
The callback can only see it if you do `[constants](...){...}`, and then it's a closure which can't be passed to a C-style function pointer.

Reply:
Fair! But then, I'd like the comment to explain that, and also for the `vector` to be cleared before each call.

(And, alternatively, `template` the function so that it can take a callable that isn't a function pointer.)

Reply:
The state-dump file callbacks also use this pattern, with the same `// `static` so `<thing>_ForEach` callback can see it` comment; what more needs explaining?

I considered clearing the vector, but this function should only ever be called once. I guess it's harmless to add though.

After this PR I'll look into how to get around the need for `static` here, e.g. `template`s or iterators or otherwise getting rid of those code-smell `ForEach` functions.